### PR TITLE
added packages/dolog.4.0.0

### PR DESCRIPTION
**Pull request by UnixJunkie**
Switched from oasis to dune.
This introduces a Dolog top namespace, hence the new major version.

### CODE REVIEW

**Comment by UnixJunkie** [comment]
I will fix all rev deps also.

**Comment by mseri** [comment]
Seems that the failure depend on external factors. Can you also have a look?

**Comment by UnixJunkie** [comment]
It would be nice to have a tool to manage constraints in opam-repository.
Here, I wanted to add an upper bound constraint, to all users of this library which
did not have one until now.

I'll have a look at the failures tomorrow.

File: packages/dolog/dolog.4.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+name: "dolog"
+authors: "Francois Berenger"
+maintainer: "unixjunkie@sdf.org"
+license: "BSD-3-Clause"
+homepage: "https://github.com/UnixJunkie/dolog"
+bug-reports: "https://github.com/UnixJunkie/dolog/issues"
+dev-repo: "git+https://github.com/UnixJunkie/dolog.git"
+depends: [
+  "ocaml"
+  "dune"
```
Comment (kit-ty-kate) [inline packages/dolog/dolog.4.0.0/opam]:
```suggestion
  "dune" {>= "1.11"}
```
The `dune-project` file in your repository says this package requires at least dune 1.11

File: packages/dolog/dolog.4.0.0/opam [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "base-unix"
+]
+build: ["jbuilder" "build" "-p" name "-j" jobs]
```
Comment (kit-ty-kate) [inline packages/dolog/dolog.4.0.0/opam]:
```suggestion
build: ["dune" "build" "-p" name "-j" jobs]
```
I think you're using dune not jbuilder

**Comment by kit-ty-kate** [comment]
> It would be nice to have a tool to manage constraints in opam-repository.
> Here, I wanted to add an upper bound constraint, to all users of this library which
> did not have one until now.

For simple things you can use `opam admin add-constraint`. In your case you could have done `opam admin add-constraint "dolog<4.0.0"`. However I've only seen a few people use it and i haven't used it myself until a few minutes ago just to try. In particular it seems to reformat all packages and it might be very annoying for review (or commit if you decide to clean it before)

**Comment by avsm** [comment]
I use `opam admin add-constraint` fairly often. The trick is to `git add` just the packages you want, and then to revert all the other constraints it adds (since you can't select individual packages with it until opam 2.1)

**Comment by UnixJunkie** [comment]
opam-admin is not in opam?! :disappointed:

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**